ops/testing.py

```diff
@@ -2254,7 +2254,17 @@ def status_get(self, *, is_app: bool = False):
         else:
             return self._unit_status
 
-    def status_set(self, status: '_StatusName', message: str = '', *, is_app: bool = False):
+    def status_set(
+        self,
+        status: "_StatusName",
+        message: str = "",
+        *,
+        is_app: bool = False,
+        by_testing_backend: bool = False,
+    ):
+        if not by_testing_backend and status in {model.ErrorStatus.name, model.UnknownStatus.name}:
```

Comment:
Statuses aren't hashable, so using a set here doesn't gain us anything (and sends the wrong signal IMO). Let's prefer a list:

```suggestion
        if status in [model.ErrorStatus.name, model.UnknownStatus.name]:
```

Reply:
Good point, updated thank you!